classes/local/apibridge.php

```diff
@@ -1791,4 +1849,226 @@ public function get_duplicated_episodeid($workflowid) {
+                        'action' => $action);
+                    
+                    // Check if the role object already exists in the acl list.
+                    $existingacl = array_filter($acl, function($v, $k) use ($roleobject) {
+                        if ($v->role == $roleobject->role && $v->action == $roleobject->action) {
+                            return true;
+                        }
+                    },ARRAY_FILTER_USE_BOTH);
+
+                    // In case the role object is new, we add it to the acl list. This helps making a clean list.
+                    if (empty($existingacl)) {
+                        $acl[] = $roleobject;
+                    }
+                }
+            }
+        }
+
+        // Put everything in params value as a string.
+        $params['acl'] = json_encode(array_values($acl));
+
+        // When there is nothing to change.
+        if ($params['acl'] == ($jsonacl)) {
+            return true;
+        }
+
+        // Update the acls via put request.
+        $api = new api();
+        $api->oc_put($resource, $params);
+
+        // Finally we return the result of that request to the server.
+        return ($api->get_http_code() == 200);
+    }
+
+    /**
+     * Make Event visible for the new course that has been imported to, by changing the ACLs.
+     *
+     * @param string $identifier event identifier
+     * @return int $courseid id of the course being imported to.
+     */
+    private function imported_events_acl_change($identifier, $courseid) {
+        // Use try to catch unwanted errors.
+        try {
+            // Make it visible to the course does the ACL change accordingly.
+            $visibilychanged = $this->change_visibility($identifier, $courseid, block_opencast_renderer::VISIBLE);
+            // In order to resolve the return result of the change_visibilty method, we assume non (false) values as true.
+            return ($visibilychanged !== false) ? true : false;
+        } catch (\moodle_exception $e) {
+            // It is unlikely, but who knows.
+            return false;
+        }
+    }
+
+    /**
+     * Map the seriesid to the course that has been imported to, by assinging the series as secondary.
+     *
+     * @return int $courseid id of the course being imported to.
+     * @param string $seriesid series id.
+     */
+    private function map_imported_series_to_course($courseid, $seriesid) {
+        try {
+            // Get the current record.
+            $mapping = seriesmapping::get_record(array('courseid' => $courseid, 'series' => $seriesid, 'isdefault' => '0'));
+
+            // If the mapping record does not exists, we create one.
+            if (!$mapping) {
+                $mapping = new seriesmapping();
+                $mapping->set('courseid', $courseid);
+                $mapping->set('series', $seriesid);
+                $mapping->set('isdefault', 0); // It is not the default series of this course.
```

Comment:
I think it would be good if the series is set as default series if we do not have a (default) series in the target course yet. It should not be possible to have a course without a default series if there are series associated with this course. So, if the target course was just created (=no default series exists) and the teacher imports the series, the series should be set as default.